### ISSUE TITLE
Why there are 2 kinds of Vector3(rp3d::Vector3 and openglframework::Vector3), any convinient way to transform? or how to do dot product between 2 kinds of vector? 

### ISSUE THREAD

**Issue by cow8** [DanielChappuis/reactphysics3d#111]
Why there are 2 kinds of Vector3(rp3d::Vector3 and openglframework::Vector3), any convinient way to transform? or how to do dot product between 2 kinds of vector?  

Currently, what I am doing is manually transform one kind into the other by constructor function:
```
#given v1,v2
rp3d::Matrix3x3 v1(1,2,3);
openglframework::Vector3 v2(1,2,3);
openglframework::Matrix3x3 v1_new(v1.x,v1.y,v1.z);
openglframework::Vector3 result=v1_new*v2;
```

it's usually the case that I want to calculate transform related to camera transform, since mCamera.getTransformMatrix() return openglframework::Matrix3x3 and rp3d::Transform() constructor take rp3d::Matrix3 and rp3d::Vector3 as input.

**Comment by DanielChappuis** [comment]
The official structure for Vector3 in ReactPhysics3D is **rp3d::Vector3**. The structure openglframework::Vector3 is the Vector3 class of the graphics engine that I use for rendering of the testbed application. This structure is not part of the ReactPhysics3D library.

If you have your own 3D simulation project, you probably already have a structure for 3D vectors. If you want to use ReactPhysics3D in your project, you need to use rp3d::Vector3 structs when you use methods of the library.
Of course at some point, you will need to translate your custom 3D vectors to the rp3d::Vector3 of ReactPhysics3D.

That's exactly what I am doing with the testbed application. It can be considered as a custom project that uses ReactPhysics3D library. The openglframework::Vector3 struct is used only in the testbed application project and when necessary I translate it to rp3d::Vector3 structs when I need to use a method of the ReactPhysics3D library.

To sum up, in your project you need to use rp3d::Vector3 when you use methods of the ReactPhysics3D library but you should not used openglframework::Vector3 in your project because it is not part of the library but only exists in the testbed application. 

However, maybe your are trying to modify the code of the ReactPhysics3d testbed application (I don't really know why). Therefore, you need to use openglframework::Vector3 for rendering code in the testbed application and rp3d::Vector3 for code related to the ReactPhysics3D library.

**Comment by cow8** [comment]
Thanks for your quick reply and also for the very nice library. And I am exactly learning how to use this library by trying to modify the testbed code. 
And now I understand the reason why we have two Vector3 after your explanation.
thanks again :  )

**Comment by DanielChappuis** [comment]
You are welcome
